internal/credential/rotate.go

```diff
@@ -161,19 +158,29 @@ func (c *Config) ValidateServiceAccountKey(
 	var validatePermissionsErr error
 	var validationCallbackErr error
 
-	for start := time.Now(); time.Since(start) < validateServiceAccountKeyTimeout; time.Sleep(200 * time.Millisecond) {
-		_, validatePermissionsErr = c.ValidateIamPermissions(ctx, permissions, opts...)
+	ticker := time.NewTicker(200 * time.Millisecond)
+	defer ticker.Stop()
 
-		if validateCredsCallback != nil {
-			validationCallbackErr = validateCredsCallback(c, opts...)
-		}
+	timeout := time.After(c.validateServiceAccountKeyTimeout)
```

Comment:
As an alternative to this, what do you think about creating a new context bounded by this timeout? That removes one of the cases below
```suggestion
	var cancel context.CancelFunc
	ctx, cancel = context.WithTimeout(ctx, c.validateServiceAccountKeyTimeout)
```